app/models/event.rb

```diff
@@ -22,6 +22,25 @@ class Event < ActiveRecord::Base
   has_many :date_ranges
   accepts_nested_attributes_for :date_ranges
 
+  # Setter for max_participants
+  # @param [Int Float] the max number of participants for the event or infinity if it is not limited
+  # @return none
+  def max_participants=(value)
+    if value == Float::INFINITY
+      self[:participants_are_unlimited] = true
+    else
+      self[:participants_are_unlimited] = false
+      self[:max_participants] = value
+    end
+  end
+
+  # Getter for max_participants
+  # @param none
+  # @return [Int Float] the max number of participants for the event or infinity if it is not limited
+  def max_participants
+    participants_are_unlimited ? Float::INFINITY : self[:max_participants]
```

Comment:
self[:max_participants] directly accesses the model attribute. It would be an infinite loop, if they would call max_participants (which is usally an implicit getter, but here overridden)